inc/Editor/DataBinding/BlockBindings.php

```diff
@@ -204,18 +204,24 @@ private static function execute_queries( array $block_context, array $source_arg
 		}
 	}
 
-	public static function should_render_empty_result( WP_Block $block ): bool {
+	public static function get_empty_or_error_state_for_block( WP_Block $block ): string|null {
```

Comment:
I would suggest keeping it as a `bool` return and inspecting block attributes
```suggestion
	public static function should_render_fallback_content( array $attributes ): bool {
```

Reply:
@ingeniumed Can you add some unit tests for this method?

Reply:
I had done it this way at first, as I wasn't 100% if the variation would load this way. Didn't revisit this after it started working, so it's simplified now.

Reply:
Yip, added it in. Got used to working on the pieces where it's not possible to add tests and missed out on this one.